#!/usr/bin/env bash
#SBATCH --cpus-per-task=16
#SBATCH --mem=48G
#SBATCH --time=18:00:00
#SBATCH --job-name=bowtie2_flye
#SBATCH --mail-user=xiaoyue.deng@students.unibe.ch
#SBATCH --mail-type=end
#SBATCH --output=/data/users/xdeng/assembly_annotation_course/outputs_errors/output_bowtie2_polishing_%j.o
#SBATCH --error=/data/users/xdeng/assembly_annotation_course/outputs_errors/error_bowtie2_polishing_%j.e
#SBATCH --partition=pall

#Add the modules
    module add UHTS/Aligner/bowtie2/2.3.4.1
    module add UHTS/Analysis/samtools/1.10

#assembly names
    #assembly_name=canu
    assembly_name=flye
course_dir=/data/users/xdeng/assembly_annotation_course
raw_data_dir=${course_dir}/participant_4
polish_evaluation_dir=${course_dir}/polish_eval
polish_dir=${polish_evaluation_dir}/polish
index_dir=${polish_dir}/index
assembly_index_dir=${index_dir}/${assembly_name}
align_dir=${polish_dir}/align
assembly_align_dir=${align_dir}/${assembly_name}
pilon_dir=${polish_dir}/pilon
assembly_pilon_dir=${pilon_dir}/${assembly_name}

#Input
#assembly=${course_dir}/canu_3/pacbio_canu_3.contigs.fasta
    assembly=${course_dir}/flye/assembly.fasta

#Run bowtie2 to align the reads to the assembly
#Create index
# bowtie2-build ${assembly} index_bowtie2_$assembly_name
                #"${assembly}"
                #"-index_bowtie2"
        
        #Index is built in folder where the script is run, move the files to the desired output folder
        #mv data/users/xdeng/assembly_annotation_course/*.bt2 ${assembly_index_dir}
                

    #Execute bowtie2
        bowtie2 --sensitive-local -x ${assembly_index_dir}/index_bowtie2 -1 ${raw_data_dir}/Illumina/*_1.fastq.gz -2 ${raw_data_dir}/Illumina/*_2.fastq.gz -S $SCRATCH/${assembly_name}.sam
                #"--sensitive-local":use the "sensitive" preset but perform local alignment.
                #"-x":specifies the basename of the index for the reference genome to be used for alignment. The index files are generated by bowtie2-build and are essential for bowtie2 to perform the alignment. 
                #"-1/-2":specify the files containing the paired-end reads
                #"-S":specifies the output file for the alignments in SAM (Sequence Alignment/Map) format.

#Convert SAM to BAM
    samtools view -b $SCRATCH/${assembly_name}.sam > $SCRATCH/${assembly_name}_unsorted.bam
        #Options entered here are:
            #"view": SAM/BAM/CRAM files.
            #"-b": BAM format.
            #"-t: A tab-delimited file. 
            #"*.sam": Input file.
            #"*_unsorted.bam": Output file.

    #Sort BAM
        samtools sort -o $SCRATCH/${assembly_name}.bam $SCRATCH/${assembly_name}_unsorted.bam
            #Options entered here are:
                #"sort": Sort alignments by leftmost coordinates, or by read name when -n is used. 
                #"-o": Write the final sorted output to file.
                #"*_unsorted.bam": Input file

    #Index BAM
        samtools index $SCRATCH/${assembly_name}.bam
            #Options entered here are:
                #"index": Index coordinate-sorted BGZIP-compressed SAM, BAM or CRAM files for fast random access.
    
    #Move the bams created in the temporary folder $SCRATCH to the output folder
                mv $SCRATCH/${assembly_name}.bam* ${assembly_align_dir}

    #Run pilon to polish the assemblies
    java -Xmx45g -jar /mnt/software/UHTS/Analysis/pilon/1.22/bin/pilon-1.22.jar \
    --genome ${assembly} --frags ${polish_dir}/align/${assembly_name}/${assembly_name}.bam --output ${assembly_name} --outdir ${assembly_pilon_dir}
        #Options entered here are:
            #"--gemone": specifies the input genome assembly file 
            #"--frags":  requires a BAM file that contains aligned paired-end reads (fragments) to the reference genome 
            #"--output": Prefix for output files
            #"--outdir": Use this directory for all output files.
